two-factor.php

```diff
@@ -66,7 +66,7 @@ function wpcom_vip_should_force_two_factor() {
 		return false;
 	}
 
-	if ( Two_Factor_Core::is_user_using_two_factor() ) {
+	if ( wpcom_vip_is_current_user_using_two_factor() ) {
```

Comment:
Think we can follow [this](https://github.com/Automattic/vip-go-mu-plugins/blob/master/two-factor.php#L137) to prevent recursion issue [mentioned here](https://beebee8p2.wordpress.com/2020/02/13/invalid-user-id-error-while-editing-user-profile-fields/#comment-18360)
```suggestion
      if ( apply_filters( 'wpcom_vip_is_current_user_using_two_factor', false ) ) {
          return false;
      }
```

And then [here](https://github.com/Automattic/vip-go-mu-plugins/blob/master/two-factor.php#L141) add:

```
        $is_current_user_using_two_factor = wpcom_vip_is_current_user_using_two_factor();

        add_filter( 'wpcom_vip_is_current_user_using_two_factor', function() use ( $is_current_user_using_two_factor ) {
            return $is_current_user_using_two_factor;
        }, 9 );
```

That way we only calculate this once *and* we avoid calculating during `map_meta_cap`

Reply:
@cojennin Can you please make these change as an extra commit on the branch that this PR uses?

Reply:
@GaryJones all set

Reply:
@cojennin just to confirm, the main reason we need both the static var and the filter is to avoid calling `wpcom_vip_is_current_user_using_two_factor` in `map_meta_cap`?

Reply:
@mjangda static var can be removed, but figured it wasn't doing any harm to limit the number of calls made by `is_user_using_two_factor`

Reply:
@mjangda removed the static var in the latest push